### ISSUE TITLE
Async parsers and serializers

### ISSUE THREAD

**Issue by damooo** [pchampin/sophia_rs#162]
@Tpt has released new crates `oxttl`, `oxxml` for parsers and serializers. 

They are async friendly. It would be great to have sophia support async parsers and serializers too. Enables proper structural concurrency in applications, integrates well with wasm+js.

Also, oxttl supports N3 too. So the migration will be much beneficial.

**Comment by pchampin** [comment]
yes, I plan to do the move towards oxttl and oxxml eventually. Setting this to N+2 (0.10) milestone for now, until I assess how much of a refactoring is required. It may be earlier it it proves not too hard.
